### PR TITLE
Rename dynamical submodules and components

### PR DESCRIPTION
**Pull request by eb8680**
This renaming-only refactoring PR renames a number of components of `chirho.dynamical` and simplifies the file structure:
- `SimulatorEventLoop` renamed to `InterruptionEventLoop`
- `DynamicTrace` renamed to `LogTrajectory`
- `DynamicTrace.trace` attribute renamed to `trajectory`
- `chirho.dynamical.handlers.interruptions.interruptions` and `chirho.dynamical.handlers.interruptions.array_observation` collapsed into a single file `chirho.dynamical.handlers.interruptions`
- `chirho.dynamical.ops.dynamical` moved up into a single file `chirho.dynamical.ops`
- `chirho.dynamical.handlers.dynamical` renamed to `chirho.dynamical.handlers.event_loop`
- `chirho.dynamical.handlers.trace` renamed to `chirho.dynamical.handlers.trajectory`
- `chirho.dynamical.internals.backend` renamed to `chirho.dynamical.internals.solver`
- `chirho.dynamical.internals.solvers` renamed to `chirho.dynamical.internals.backends`
- `test_trace.py` renamed to `test_log_trajectory.py`
- All tests from `test_state_trajectory.py` moved into `test_log_trajectory.py`

I tried to collect all of the remaining user-facing renaming I wanted to do into a single PR to limit merge conflicts. All of the changes here are editor-assisted renaming and moving; there are no other changes to any of the code.

### CODE REVIEW

**Comment by eb8680** [comment]
The intro notebook is missing more than just these changes and should be updated separately.

File: tests/dynamical/test_state_trajectory.py [hunk 1]
```diff
@@ -1,23 +0,0 @@
-import logging
```
Comment (SamWitty) [inline tests/dynamical/test_state_trajectory.py:1]: Minor nit: I think this test should be removed in a subsequent PR that merges `Trajectory` and `State`.

**Review by SamWitty — CHANGES_REQUESTED** [review]
Looks good overall. Would prefer to keep removal of (potentially unnecessary) test separate.

**Review by SamWitty — APPROVED** [review]
Ignore last review. Missed that the trajectory tests were consolidated.

**Pull request merged**